packages/data/src/redux-store/index.js

```diff
@@ -387,17 +398,38 @@ function mapResolvers( resolvers, selectors, store, resolversCache ) {
 
 		const selectorResolver = ( ...args ) => {
 			async function fulfillSelector() {
-				const state = store.getState();
+				if ( resolversCache.isRunning( selectorName, args ) ) {
+					registry.__unstableSuspend(
+						registry.resolveSelect(
```

Comment:
This looks like potential infinite recursion? Calling a selector starts to resolve it, calling the resolver, then the resolver calls the selector again, triggering another resolution again? Maybe it's all carefully orchestrated so that the recursion always ends quickly, just flagging this as something that's worth a thorough review and testing.